test/testdata/rewriter/private_with_bad_default.rb

```diff
@@ -0,0 +1,13 @@
+# typed: strict
+
+class Main
+  extend T::Sig
+
+  sig {params(arg0: T.nilable(String)).returns(NilClass)}
+  private def private_with_bad_default(arg0: 0) # error: Argument does not have asserted type `T.nilable(String)`
+  end
+
+  sig {params(arg0: T.nilable(String)).returns(NilClass)}
+  def public_with_bad_default(arg0: 0) # error: Argument does not have asserted type `T.nilable(String)`
+  end
+end
```

Comment:
I tested that this failed on master (since DefaultArgs was not seeing behind the `private` call, it wasn't creating `private_with_bad_default<defaultArg>1` with a corresponding sig, which was then likewise not tripping the part of the pipeline that emits an error when a default doesn't match the sig.